data/tutorials/guides/0tt_05_s_expressions.md

```diff
@@ -0,0 +1,267 @@
+# Std.array_of_sexp Std.int_of_sexp (Sexp.List [Sexp.Atom "3"; Sexp.Atom "2"; Sexp.Atom "1"]);;
+- : int array = [|3; 2; 1|]
+```
+
+### Setting Up `ppx_sexp_conv`
+
+[`ppx_sexp_conv`](https://github.com/janestreet/ppx_sexp_conv) is a library and a [PPX rewriter](https://ocaml.org/docs/metaprogramming#ppx-rewriters) that allows conversions between S-Expressions and OCaml types.
+
+First, `ppx_sexp_conv` needs to be installed:
+```shell
+$ opam install ppx_sexp_conv
+```
+
+Then, to use `ppx_sexp_conv` in your project, you must list it as a dependency and as a preprocessor in your `dune` file:
+```lisp
+(library
+  (name <your_lib>)
+ (preprocess
+  (pps ppx_sexp_conv))
+ (libraries ppx_sexp_conv <your_dependencies>))
+```
+
+Finally, to add the predefined converters to the scope, you must open the `Sexplib.Std` module.
+
+### Automatically Generate Conversion Functions
+
+`ppx_sexp_conv` will automatically generate conversion functions for the selected OCaml types.
+
+These functions are of the form:
+- `sexp_of_*` to convert from an OCaml type to a S-Expression
+- `*_of_sexp` to convert from a S-Expression to an OCaml type
+
+For example, `sexp_of_int` and `int_of_sexp` are generated for basic the `int` type. These functions will be specifically named after the OCaml type they allow users to convert to and from.
+
+To automatically generate such functions for your own type, you must annotate the type definition with an attribute: `[@@deriving sexp]`, such as:
+
+```ocaml
+# type color = Red | Blue | Green [@@deriving sexp];;
+type color = Red | Blue | Green
+val color_of_sexp : Sexp.t -> color = <fun>
+val sexp_of_color : color -> Sexp.t = <fun>
+```
+
+This will prompt the `ppx_sexp_conv` preprocessor to generate both `sexp_of_color` and `color_of_sexp` functions, to convert between values of the type `color` and S-Expressions.
+
+You can then use the generated function as usual:
+```ocaml
+# sexp_of_color Blue;;
+- : Sexp.t = Sexplib.Sexp.Atom "Blue"
+```
+
+To only generate the `sexp_of_color` function, you can annotate the type with `[@@deriving sexp_of]`. Likewise, to only generate the `color_of_sexp` function, you can annotate the type with `[@@deriving of_sexp]`.
+
+Every component of your type must also be converted:
+
+```ocaml
+# type direction = Up | Bottom | Left | Right [@@deriving sexp];;
+type direction = Up | Bottom | Left | Right
+val direction_of_sexp : Sexp.t -> direction = <fun>
+val sexp_of_direction : direction -> Sexp.t = <fun>
+
+# type movement = { direction : direction; distance: int }
+[@@deriving sexp];;
+type movement = { direction : direction; distance : int; }
+val movement_of_sexp : Sexp.t -> movement = <fun>
+val sexp_of_movement : movement -> Sexp.t = <fun>
+```
+
+In the example above, annotating the type `movement` is not enough. It depends on the type `direction`, so `direction` must also be annotated with `[@@deriving sexp]` for the `sexp_of_movement` and `movement_of_sexp` functions to be generated.
+
```

Comment:
Maybe add a few words about error handling by giving a pointer even when not discussing it here.